asyn/drvAsynSerial/drvAsynIPServerPort.c

```diff
@@ -324,6 +324,12 @@ static void connectionListener(void *drvPvt)
 
         } else {
             clientFd = epicsSocketAccept(tty->fd, (struct sockaddr *) &clientAddr, &clientLen);
+            if (tty->fd == INVALID_SOCKET) {
+                asynPrint(pasynUser, ASYN_TRACE_FLOW,
+                    "drvAsynIPServerPort: terminating connection thread for %s\n",
+                    tty->serverInfo);                
+                break; /* we must be in ioc shutdown and ttyCleanup has been called */
```

Comment:
... or maybe the file descriptor limit was reached and accept() has returned with `errno==EMFILE`.

Reply:
Do we need to add a separate test for this condition when we check `clientFd` at https://github.com/FreddieAkeroyd/asyn/blob/19db7d33e5576393f99feeb196f88b7d321e5a95/asyn/drvAsynSerial/drvAsynIPServerPort.c#L336  or would this situation also affect `tty->fd` ?